### PR TITLE
CRIMAPP-876 add validation for employed client journey

### PR DESCRIPTION
**Pull request by RoseSAK**
## Description of change

Add validation for employed client journey, and specs

## Link to relevant ticket

[CRIMAPP-876](https://dsdmoj.atlassian.net/browse/CRIMAPP-876)

## How to manually test the feature

[CRIMAPP-876]: https://dsdmoj.atlassian.net/browse/CRIMAPP-876?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/validators/employment_details/answers_validator.rb [hunk 1]
```diff
@@ -39,6 +43,55 @@ def not_working_details_complete?
...
+    def validate_employment
+      return unless employed?
+
+      validate_employment_details
+      validate_employment_income
+    end
+    # :nocov:
+
+    def validate_employment_details
+      return unless requires_full_means_assessment?
```
Comment (arthurashman) [inline app/validators/employment_details/answers_validator.rb:56]: Is this the trigger for collecting full employment details? Is it not just `income_above_threshold?`

File: app/models/concerns/type_of_employment.rb [hunk 2]
```diff
@@ -7,6 +7,10 @@ def not_working?
     income.employment_status.include?(EmploymentStatus::NOT_WORKING.to_s)
   end
 
+  def employed?
+    income.employment_status.include?(EmploymentStatus::EMPLOYED.to_s)
+  end
```
Comment (arthurashman) [inline app/models/concerns/type_of_employment.rb:12]: I think we should make this EMPLOYED only. As if employement_status = ['employed', 'self-employed'] I think the route and therefore validations is unique

File: spec/validators/employment_details/answers_validator_spec.rb [hunk 3]
```diff
@@ -87,23 +97,276 @@
...
+        let(:employments) { [instance_double(Employment, complete?: true)] }
+
+        it 'adds an error to :other_work_benefit' do
+          expect(errors).to receive(:add).with(:applicant_other_work_benefit_received, :incomplete)
+
+          validate
+        end
+      end
+
+      context 'when all requires attributes are present and complete' do
```
Comment (arthurashman) [inline spec/validators/employment_details/answers_validator_spec.rb:160]: *required

**Review by arthurashman — APPROVED** [review]
👍

File: app/models/concerns/type_of_employment.rb (hunk 2 above)
Comment (RoseSAK) [reply app/models/concerns/type_of_employment.rb:12]: Decided to leave it for now and update when we get around to self employed journey

File: app/validators/employment_details/answers_validator.rb [hunk 4]
```diff
@@ -13,21 +14,24 @@ def initialize(record)
...
 
-      not_working_details_complete?
+      errors.add :partner_employment_status, :incomplete if record.partner_employment_status.blank?
+    end
+
+    def applicable?
+      requires_means_assessment?
     end
 
     def not_working_details_complete?
```
Comment (ripan) [inline app/validators/employment_details/answers_validator.rb:37]: @RoseSAK What do you think of changing method name
```suggestion
def working_details_complete?
  income.date_job_lost.blank?
end
```

**Pull request merged**